src_tests/test_exact_distribution_calculator.py

```diff
@@ -0,0 +1,47 @@
+import unittest
+from numpy import array, std
+
+from src.ExactLossyBosonSamplingDistributionCalculator.ExactLossyBosonSamplingDistributionCalculator \
+    import ExactLossyBosonSamplingDistributionCalculator, BosonSamplingExperimentConfiguration
+
+
+class TestExactLossyBosonSamplingDistributionCalculator(unittest.TestCase):
+
+    def setUp(self) -> None:
+        # Define some additional variables for more clear experiment configuration assignment.
+        self.number_of_particles_lost = 2
+
+        # Generate permutation matrix and define initial state.
+        self.permutation_matrix = array([
+            [0, 0, 1, 0, 0],
+            [1, 0, 0, 0, 0],
+            [0, 0, 0, 1, 0],
+            [0, 0, 0, 0, 1],
+            [0, 1, 0, 0, 0],
+        ])
+
+        self.initial_state = [1, 1, 1, 0, 0]
+
+        # Create configuration object.
+        self.experiment_configuration = BosonSamplingExperimentConfiguration(
+            interferometer_matrix=self.permutation_matrix,
+            possible_outcomes=[[1, 0, 0, 0, 0], [0, 1, 0, 0, 0], [0, 0, 0, 0, 1]],
+            initial_state=self.initial_state,
+            number_of_modes=len(self.initial_state),
+            initial_number_of_particles=sum(self.initial_state),
+            number_of_particles_lost=self.number_of_particles_lost,
+            number_of_particles_left=sum(self.initial_state) - self.number_of_particles_lost
+        )
+
+    def test_probabilities_sum(self) -> None:
+        exact_distribution_calculator = ExactLossyBosonSamplingDistributionCalculator(self.experiment_configuration)
+        exact_distribution = exact_distribution_calculator.calculate_exact_distribution()
+        self.assertAlmostEqual(sum(exact_distribution), 1.0, delta=0.0001)
```

Comment:
Nitpick: Have you considered using `1e-4` instead of `0.0001`? It's easier to read for me.

Reply:
+1

Reply:
Usually I tend to do it from 1e-5 and onward, but you're right. xD